crates/algokit_utils/src/transactions/sender.rs

```diff
@@ -372,31 +373,54 @@ impl TransactionSender {
             }
         }
 
-        if params.creator.is_none() {
-            let asset_info = self
-                .asset_manager
-                .get_by_id(params.asset_id)
-                .await
-                .map_err(|e| {
-                    TransactionSenderError::ValidationError(format!(
-                        "Asset {} validation failed: {}",
-                        params.asset_id, e
-                    ))
-                })?;
-            let creator_address = &asset_info.creator;
-            params.creator = Some(Address::from_str(creator_address).map_err(|e| {
-                TransactionSenderError::InvalidParameters(format!(
-                    "Invalid creator address for asset {}: {}",
-                    params.asset_id, e
-                ))
-            })?);
-        }
-
         let mut composer = self.new_group();
         composer.add_asset_opt_out(params)?;
         self.send_and_parse(composer, send_params).await
     }
 
+    /// Send asset opt-out transaction with automatic creator resolution.
+    /// This is a convenience method that automatically resolves the asset creator
+    /// for backward compatibility with code that doesn't explicitly provide the creator.
+    pub async fn asset_opt_out_with_auto_creator(
```

Comment:
I'm not sure we should have this method.
Maybe we go back to having the close_remainder_to field optional and resolve that to the creator address if it's None.